### PR TITLE
Macroquad Permissions example

### PR DESCRIPTION
**Pull request by Heezay**

### CODE REVIEW

File: examples/macroquad-ui/Cargo.toml [hunk 1]
```diff
@@ -5,16 +5,12 @@ authors = ["DodoRare Team <support@dodorare.com>"]
 edition = "2021"
 
 [dependencies]
+crossbow-permissions = { path = "../../plugins/permissions", version = "0.1.3" }
```
Comment (enfipy) [inline examples/macroquad-ui/Cargo.toml:8]: Replace this with the feature "plugin-permissions" in the crossbow package.

File: plugins/permissions/Cargo.toml [hunk 2]
```diff
@@ -9,7 +9,10 @@ license = "Apache-2.0"
...
 readme = "README.md"
 
-[dependencies]
-
 [target.'cfg(target_os = "android")'.dependencies]
+ndk-glue = "0.6.0"
+
+[dependencies]
 jni = "0.19"
+ndk-context = "0.1"
```
Comment (enfipy) [inline plugins/permissions/Cargo.toml:17]: All ndk/jni should be in target_os android deps. Move it to ndk-glue.

File: examples/macroquad-ui/src/main.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
+#[cfg(target_os = "android")]
+use crossbow_permissions::request_permission::ask_for_permission;
```
Comment (enfipy) [inline examples/macroquad-ui/src/main.rs:2]: It's better to make a single cross-platform function. Don't add `cfg target_os` here

File: plugins/permissions/Cargo.toml [hunk 4]
```diff
@@ -9,10 +9,8 @@ license = "Apache-2.0"
 keywords = ["permissions", "android", "ios"]
 readme = "README.md"
 
+# [dependencies]
 [target.'cfg(target_os = "android")'.dependencies]
 ndk-glue = "0.6.0"
-
-[dependencies]
 jni = "0.19"
-ndk-context = "0.1"
```
Comment (enfipy) [inline plugins/permissions/Cargo.toml]: No ndk-context? But we use it in `request_permission.rs`

**Review by enfipy — CHANGES_REQUESTED** [review]

File: plugins/permissions/Cargo.toml (hunk 2 above)
Comment (Heezay) [reply plugins/permissions/Cargo.toml]: Will return as was it after my refactoring done

File: examples/macroquad-ui/res/AndroidManifest.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (Heezay) [reply examples/macroquad-ui/res/AndroidManifest.xml]: Removed. Wasn't need to use it now.

File: plugins/permissions/src/lib.rs [hunk 6]
```diff
@@ -1,6 +1,6 @@
-mod android_permission;
-mod android_permission_group;
-mod request_permission;
+pub mod android_permission;
```
[the remark this answers is not in the document]
Comment (Heezay) [reply plugins/permissions/src/lib.rs:1]: Was changed.

File: examples/macroquad-ui/src/main.rs (hunk 3 above)
Comment (Heezay) [reply examples/macroquad-ui/src/main.rs:2]: Ok. It was added because of rust-analyzer issues.

File: examples/macroquad-ui/Cargo.toml (hunk 1 above)
Comment (Heezay) [reply examples/macroquad-ui/Cargo.toml:8]: Don't realize how to do this now

File: plugins/permissions/Cargo.toml (hunk 4 above)
Comment (Heezay) [reply plugins/permissions/Cargo.toml]: Yes, we need it. I thought ndk-glue contains it.

**Review by enfipy — APPROVED** [review]

**Pull request merged**